src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -820,7 +933,286 @@ internal void TakeSnapshot(bool outputDelta)
             }
         }
 
-        private void UpdateHistogram(double number)
+        internal void TakeSnapshotWithExemplar(bool outputDelta)
+        {
+            switch (this.aggType)
+            {
+                case AggregationType.LongSumIncomingDelta:
+                case AggregationType.LongSumIncomingCumulative:
+                    {
+                        var sw = default(SpinWait);
+                        while (true)
+                        {
+                            if (Interlocked.Exchange(ref this.mpComponents.IsCriticalSectionOccupied, 1) == 0)
+                            {
+                                // Lock acquired
+
+                                if (outputDelta)
+                                {
+                                    long initValue = this.runningValue.AsLong;
```

Comment:
Same issue here as mentioned in https://github.com/open-telemetry/opentelemetry-dotnet/pull/4256/files#r1124113958